lib/boombox/rtp.ex

```diff
@@ -0,0 +1,175 @@
+          | %{}
+
+  @type parsed_track_config :: %{
+          encoding_name: RTP.encoding_name(),
+          encoding_specific_params: parsed_encoding_specific_params(),
+          payload_type: RTP.payload_type(),
+          clock_rate: RTP.clock_rate()
+        }
+
+  @type parsed_in_opts :: %{
+          port: :inet.port_number(),
+          track_configs: %{audio: parsed_track_config(), video: parsed_track_config()}
+        }
+
+  @spec create_input(Boombox.in_rtp_opts()) :: Wait.t()
+  def create_input(opts) do
+    _parsed_options = validate_and_parse_options(opts)
+
+    spec =
+      child(:udp_source, %Membrane.UDP.Source{local_port_no: opts[:port]})
+      |> child(:rtp_demuxer, Membrane.RTP.Demuxer)
+
+    %Wait{actions: [spec: spec]}
+  end
+
+  @spec handle_new_rtp_stream(
+          ExRTP.Packet.uint32(),
+          ExRTP.Packet.uint7(),
+          [ExRTP.Packet.Extension.t()],
+          State.t()
+        ) :: {Wait.t() | Ready.t(), State.t()}
+  def handle_new_rtp_stream(ssrc, payload_type, _extensions, state) do
+    {:rtp, rtp_opts} = state.input
+
+    parsed_opts = validate_and_parse_options(rtp_opts)
+
+    {media_type, track_config} =
+      Enum.find(parsed_opts.track_configs, fn {_media_type, track_config} ->
+        track_config.payload_type == payload_type
+      end)
+
+    %PayloadFormat{depayloader: depayloader} = PayloadFormat.get(track_config.encoding_name)
+
+    spec =
+      case track_config.encoding_name do
+        :H264 ->
+          ppss = Map.get(track_config.encoding_specific_params, :ppss, [])
+          spss = Map.get(track_config.encoding_specific_params, :spss, [])
+
+          get_child(:rtp_demuxer)
+          |> via_out(Membrane.Pad.ref(:output, ssrc))
+          |> child({:jitter_buffer, ssrc}, %Membrane.RTP.JitterBuffer{
+            clock_rate: track_config.clock_rate
+          })
+          |> child({:rtp_depayloader, ssrc}, depayloader)
+          |> child({:rtp_in_parser, ssrc}, %Membrane.H264.Parser{ppss: ppss, spss: spss})
+
+        :AAC ->
+          audio_specific_config = track_config.encoding_specific_params.audio_specific_config
+          bitrate_mode = track_config.encoding_specific_params.bitrate_mode
+
+          get_child(:rtp_demuxer)
+          |> via_out(Membrane.Pad.ref(:output, ssrc))
+          |> child({:jitter_buffer, ssrc}, %Membrane.RTP.JitterBuffer{
+            clock_rate: track_config.clock_rate
+          })
+          |> child({:rtp_depayloader, ssrc}, struct(depayloader, mode: bitrate_mode))
+          |> child({:rtp_in_parser, ssrc}, %Membrane.AAC.Parser{
+            audio_specific_config: audio_specific_config
+          })
+          |> child(:rtp_in_aac_decoder, Membrane.AAC.FDK.Decoder)
+      end
+
+    state =
+      if state.track_builders == nil do
+        %{state | track_builders: %{media_type => spec}}
+      else
+        Bunch.Struct.put_in(state, [:track_builders, media_type], spec)
+      end
```

Comment:
Endpoints shouldn't modify the boombox state, otherwise we get a spaghetti. If you need to keep some state, add `rtp_state` field to the boombox state and use that.